agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2073,6 +2073,25 @@ bool backhaul_manager::handle_1905_1_message(ieee1905_1::CmduMessageRx &cmdu_rx,
     }
 }
 
+bool backhaul_manager::get_sta_bssid(
+    const std::unordered_map<sMacAddr, associated_clients_t> &clients_map, const sMacAddr &sta_mac,
+    sMacAddr &bssid)
+{
+    return std::any_of(clients_map.begin(), clients_map.end(),
```

Comment:
>> Can't it be done with ranged for-loops? They are more readable and have less indentation.

That they are more readable is a matter of taste.

>> It is mention as preferred on https://github.com/prplfoundation/prplMesh/blob/master/CODINGSTYLE.md#loops.

That is only compared to "normal" for-loops, not compared to std::algorithms.


> You are right Moran, but `cppcheck` thinks otherwise :/

That's not good enough of a reason. If a tool doesn't do what we want, we have to change the tool, not what we want.

That said, I guess the main argument against using any_of is that it's too verbose? There are actually two ways that we can reduce the verbosity: by using C++14 auto, and by defining wrappers for the algorithms that take a container rather than an iterator.

```
    return any_of(clients_map, [&](const auto &vap) {
        return any_of(vap.second, [&](const auto &mac) {
            if (mac.first == sta_mac) {
                bssid = vap.first;
                return true;
            }
            return false;
        });
    });
```
where
```
template <class T, class UnaryPredicate>
bool any_of(const T& container, UnaryPredicate p)
{
    return any_of(std::begin(container), std::end(container), p);
}
```

Without that helper function, it's still not that bad if you use auto:

```
    return std::any_of(clients_map.begin(), clients_map.end(), [&](const auto &vap) {
        return std::any_of(vap.second.begin(), vap.second.end(), [&](const auto &mac) {
            if (mac.first == sta_mac) {
                bssid = vap.first;
                return true;
            }
            return false;
        });
    });
```

but I agree that it becomes bad when the lambda declaration no longer fits on the same line.

So I think the indentation argument is not really valid. Thus, the question is: which of the two is more readable?

I think we'll need to take a vote on #democracy on this.